presto-main/src/main/java/com/facebook/presto/operator/project/PageFilter.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.project;
+
+import com.facebook.presto.spi.ConnectorSession;
+import com.facebook.presto.spi.Page;
+
+public interface PageFilter
+{
+    boolean isDeterministic();
+
+    InputChannels getInputChannels();
+
+    SelectedPositions filter(ConnectorSession session, Page page);
```

Comment:
Shouldn't such filter chain be merged into single filter by optimizer? If so we probably do not need to optimize for such case.